### ISSUE TITLE
animals_00_multilayer.avif needs to remove the item only containing the first layer or use `lsel`

### ISSUE THREAD

**Issue by leo-barnes** [AOMediaCodec/av1-avif#152]
@joedrago noticed in [this libavif issue](https://github.com/AOMediaCodec/libavif/pull/640) that `dav1d` didn't correctly decode the first item in `animals_00_multilayer.avif`. That item only contains the first layer out of a sequence containing two spatial layers.

Some debugging by @joedrago and spec reading by @wantehchang indicates that this item is either not correct according to the AV1 spec, or at least in the zone of undefined behaviour, since it's basically a truncated elementary stream which it's not really clear how a decoder should handle.

The purpose of `animals_00_multilayer.avif` was to verify that a multilayer file not containing any of the new boxes worked correctly, so I think the best solution would be to simply remove the truncated item rather than add `lsel`. (I guess it served it's purpose by highlighting edge cases I hadn't even thought about 😄.)

**Comment by leo-barnes** [comment]
@wantehchang 
> The image item decoded by avifdec has only one AV1 frame OBU after the AV1 sequence header OBU. That AV1 frame OBU has obu_extension_flag=0, so it does not have an obu_extension_header() and its temporal_id and spatial_id are both 0 by default. Note: Under the draft revision of the AV1 spec, this AV1 frame OBU should set obu_extension_flag to 1 and have an obu_extension_header() with temporal_id=0 and spatial_id=0. It would be good to follow the new requirement.

How would I go about correcting this? Do I just update my libaom version when encoding, or do I need to set some options when encoding?

**Comment by wantehchang** [comment]
Hi Leo,

Since you used libaom to encode the AV1 bitstream, this shows the libaom encoder has not been updated to conform to the new requirements in the draft revision of the AV1 spec. I can look into that.

An alternative is to edit that AV1 Frame OBU to set obu_extension_flag (one bit) to 1 and insert an obu_extension_header() (one byte).

**Comment by leo-barnes** [comment]
> Since you used libaom to encode the AV1 bitstream, this shows the libaom encoder has not been updated to conform to the new requirements in the draft revision of the AV1 spec. I can look into that.

Do you know if/which commit has been updated to conform to the new requirements? I used a pretty old build of libaom initially if I remember correctly, so that might be the issue.

**Comment by joedrago** [comment]
If you choose to remove the truncated item, be sure to switch the primary item (`pitm`) to the other remaining item.

**Comment by wantehchang** [comment]
@leo-barnes

Hi Leo,

Sorry about the late reply. I found that even the current version of the AV1 spec (Version 1.0.0 with Errata 1) requires the AV1 frame OBU with `temporal_id=0` and `spatial_id=0` set `obu_extension_flag` to 1. Section 7.5. Ordering of OBUs says on page 205:

> If a coded video sequence contains at least one enhancement layer (OBUs with spatial_id greater than 0 or temporal_id
> greater than 0) then all frame headers and tile group OBUs associated with base (spatial_id equals 0 and temporal_id
> equals 0) and enhancement layer (spatial_id greater than 0 or temporal_id greater than 0) data must include the OBU
> extension header.

All versions of the libaom encoder fail to meet this requirement. (This requirement is unchanged in the draft revision of the AV1 spec.)

Last week I wrote [a libaom patch](https://aomedia-review.googlesource.com/c/aom/+/139587) to implement the new requirements, which also takes care of the requirement I quoted above.

In any case, I think it is more important to remove the truncated item. We can ignore the `obu_extension_flag` issue.
